cli/azd/pkg/tools/github/github.go

```diff
@@ -223,12 +257,65 @@ func (cli *ghCli) GitHubActionsExists(ctx context.Context, repoSlug string) (boo
 	return true, nil
 }
 
+func (cli *ghCli) ForceConfigureAuth(authMode AuthTokenSource) {
```

Comment:
Will this happen per command-run?  or is it persisted to the ghCli ?

It would not be good if we are leaving the ghCli with the overridden settings.  So, hopefully this is per-command run. Otherwise, we need to undo the override on `azd pipeline config` defer

Reply:
No, it's temporal per-command run. Technically, it's per lifetime of a `ghCli` object instance -- so we can decide inside `azd` invocation.